tests/integ/test_stored_procedure.py

```diff
@@ -2501,3 +2511,72 @@ def worker_process(parquet_queue):
 
     ingestion = sproc(core_ingestion_logic, return_type=StringType())
     assert ingestion() == "success"
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="data source is not supported in local testing",
+    run=False,
+)
+def test_data_source_udtf_ingestion(session, db_parameters):
+    with Session.builder.configs(db_parameters).create() as new_session:
+        new_session.custom_package_usage_config["enabled"] = True
+        new_session.custom_package_usage_config["force_push"] = True
+        new_session.add_packages("oracledb")
+        new_session.add_packages("snowflake-snowpark-python")
```

Comment:
do we need to create a new session? or the fixture session is already good

Reply:
what would happen if you don't do these operations on the new session, would the test fail?